wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt

```diff
@@ -240,6 +242,19 @@ class MainViewModel @Inject constructor(
         }
     }
 
+    suspend fun getDeviceTimeSkew(): Long {
+        return try {
+            val systemTimeMillis = System.currentTimeMillis()
+            val result = Constants.HTTP_CLIENT.head("https://www.dash.org/")
+            val networkTime = result.headers.getDate("date")?.time
```

Comment:
Here we're comparing the device's time with the time dash.org returns in headers.

Initially, I tried to request time from an NTP server and parse the binary response with some ChatGPT-assisted code, but it proved to be not reliable. For a more reliable solution, we would probably need to use a dedicated NTP library, but it might be overkill.